internal/store/models/storedobject.go

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// models describes the data types that will be used when storing export data.
+package models
+
+// StoredObject is the atomic and most abstract description of what is collected by the export store system.
+type StoredObject struct {
+	// ID is the unique identifier for this record.
```

Comment:
Please add " once stored" since it is empty until it is persisted.

Reply:
Should `ID` be an unexported field so that it can only be modified by the `Store` function?

Reply:
I added a commit with a constructor. This has the benefit of not allowing `ID` or `RetryCount` to be editable by an end user. Is that the approach we should go with?

Reply:
They need to be exported so they can be read.